xcdat/regridder/xgcm.py

```diff
@@ -0,0 +1,238 @@
+from typing import Any, Dict, Literal, Optional, Union, get_args
+
+import xarray as xr
+from xgcm import Grid
+
+from xcdat.logger import setup_custom_logger
+from xcdat.regridder.base import BaseRegridder, preserve_bounds
+
+XGCMVerticalMethods = Literal["linear", "conservative", "log"]
+
+logger = setup_custom_logger(__name__)
+
+
+class XGCMRegridder(BaseRegridder):
+    def __init__(
+        self,
+        input_grid: xr.Dataset,
+        output_grid: xr.Dataset,
+        method: XGCMVerticalMethods = "linear",
+        target_data: Optional[Union[str, xr.DataArray, None]] = None,
```

Comment:
```suggestion
        target_data: Optional[Union[str, xr.DataArray]] = None,
```

Remove `None` in type annotation since `Optional` implies it

Reply:
Resolved with 7420928bfa6cccc66c9db25d43a6a21a4814e31f.